src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -940,85 +982,85 @@ private async Task SendRstStreamAsync(int streamId, Http2ProtocolErrorCode error
                 (buffer.Slice(0, maxSize), buffer.Slice(maxSize)) :
                 (buffer, Memory<byte>.Empty);
 
-        private void WriteIndexedHeader(int index)
+        private void WriteIndexedHeader(int index, ref ArrayBuffer headerBuffer)
```

Comment:
Seems like these methods could be moved to `HttpStream` now that they don't operate on the connection level.

Reply:
They're only used by Http2Connection.SendAsync, and in some cases before we actually have an Http2Stream instance, so they'd need to be static, but that would then cause issues for Trace.